homeassistant/components/xiaomi_miio/air_quality.py

```diff
@@ -0,0 +1,193 @@
+
+    try:
+        device = AirMonitorB1(name, AirQualityMonitor(host, token, model=None))
+
+    except DeviceException:
+        raise PlatformNotReady
+
+    hass.data[DATA_KEY][host] = device
+    async_add_entities([device], update_before_add=True)
+
+
+class AirMonitorB1(AirQualityEntity):
+    """Air Quality class for Xiaomi cgllc.airmonitor.b1 device."""
+
+    def __init__(self, name, device):
+        """Initialize the entity."""
+        self._name = name
+        self._device = device
+        self._icon = "mdi:cloud"
+        self._manufacturer = "Xiaomi"
+        self._unit_of_measurement = "μg/m3"
+        self._model = None
+        self._mac_address = None
+        self._sw_version = None
+        self._carbon_dioxide_equivalent = None
+        self._relative_humidity = None
+        self._particulate_matter_2_5 = None
+        self._temperature = None
+        self._total_volatile_organic_compounds = None
+
+    async def async_update(self):
+        """Fetch state from the miio device."""
+
+        try:
+            if self._model is None:
+                info = await self.hass.async_add_executor_job(self._device.info)
+                self._model = info.model
+                self._mac_address = info.mac_address
+                self._sw_version = info.firmware_version
+
+            state = await self.hass.async_add_executor_job(self._device.status)
+            _LOGGER.debug("Got new state: %s", state)
+
+            self._carbon_dioxide_equivalent = state.co2e
+            self._relative_humidity = round(state.humidity, 1)
+            self._particulate_matter_2_5 = round(state.pm25, 1)
+            self._temperature = round(state.temperature, 1)
+            self._total_volatile_organic_compounds = round(state.tvoc, 3)
+
+        except DeviceException as ex:
+            _LOGGER.error("Got exception while fetching the state: %s", ex)
+
+    @property
+    def name(self):
+        """Return the name of this entity, if any."""
+        return self._name
+
+    @property
+    def device(self):
+        """Return the name of this entity, if any."""
+        return self._device
+
+    @property
+    def icon(self):
+        """Return the icon to use for device if any."""
+        return self._icon
+
+    @property
+    def manufacturer(self):
+        """Return the manufacturer version."""
+        return self._manufacturer
+
+    @property
+    def model(self):
+        """Return the device model."""
+        return self._model
+
+    @property
+    def sw_version(self):
+        """Return the software version."""
+        return self._sw_version
+
+    @property
+    def mac_address(self):
+        """Return the mac address."""
+        return self._mac_address
+
+    @property
+    def unique_id(self):
+        """Return the unique ID."""
+        return f"{self._model}-{self._mac_address}"
+
+    @property
+    def carbon_dioxide_equivalent(self):
```

Comment:
Can we implement `carbon_dioxide` property instead?

Reply:
As far as I understand there is a difference between co2e an co2. co2e is more precise in this case.